### PR TITLE
perf: enable local cache and use `tablepool` to reuse the temporary table

### PR DESCRIPTION
**Pull request by membphis**
this PR depends on: https://github.com/apache/skywalking-nginx-lua/pull/50

### CODE REVIEW

**Comment by wu-sheng** [comment]
@membphis When do you plan to finish this?

**Comment by membphis** [comment]
I will continue this PR from January 1st to 2nd next year

**Comment by membphis** [comment]
the test result is coming ^_^

**Comment by membphis** [comment]
The test result obtained when the skywalking service is not started. 

```shell
# this PR
➜  skywalking-nginx-lua git:(perf) ✗ wrk http://127.0.0.1:8080/ingress
Running 10s test @ http://127.0.0.1:8080/ingress
  2 threads and 10 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   738.99us  415.02us   5.29ms   92.24%
    Req/Sec     7.30k     1.77k    8.27k    88.12%
  146615 requests in 10.10s, 36.77MB read
Requests/sec:  14516.75
Transfer/sec:      3.64MB

# master branch
➜  skywalking-nginx-lua-upstream git:(master) ✗ wrk http://127.0.0.1:8080/ingress
Running 10s test @ http://127.0.0.1:8080/ingress
  2 threads and 10 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   801.90us  459.00us   4.60ms   91.62%
    Req/Sec     6.74k     1.71k    7.68k    87.13%
  135492 requests in 10.10s, 33.98MB read
Requests/sec:  13416.06
Transfer/sec:      3.36MB
```

**Comment by membphis** [comment]
The test result obtained when the skywalking service is running. 

```shell
# current PR
➜  skywalking-nginx-lua git:(perf) ✗ wrk http://127.0.0.1:8080/ingress
Running 10s test @ http://127.0.0.1:8080/ingress
  2 threads and 10 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   697.70us  248.50us   2.57ms   90.55%
    Req/Sec     7.26k     0.92k    8.39k    51.00%
  144390 requests in 10.00s, 36.21MB read
Requests/sec:  14436.94
Transfer/sec:      3.62MB

# master branch
➜  skywalking-nginx-lua-upstream git:(master) ✗ wrk http://127.0.0.1:8080/ingress
Running 10s test @ http://127.0.0.1:8080/ingress
  2 threads and 10 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   738.83us  275.64us   3.41ms   89.22%
    Req/Sec     6.86k     0.93k   11.76k    82.59%
  137225 requests in 10.10s, 34.41MB read
Requests/sec:  13586.21
Transfer/sec:      3.41MB
```

**Comment by moonming** [comment]
This PR contains some code style changes that have nothing to do with this PR's title

File: lib/skywalking/util.lua [hunk 1]
```diff
@@ -119,4 +119,50 @@ else
 end
 
 
+if _M.is_ngx_lua then
```
Comment (wu-sheng) [inline lib/skywalking/util.lua:122]: Could we provide a fall back to no-pool mechanism? But pool could be default. If there is a bug of this implementation, we have a chance to disable this.

**Comment by membphis** [comment]
> This PR contains some code style changes that have nothing to do with this PR's title

the others are mainly about local cache, so it is fine.

File: lib/skywalking/util.lua (hunk 1 above)
Comment (membphis) [reply lib/skywalking/util.lua:122]: added, please take a look when you have time

**Review by wu-sheng — APPROVED** [review]

**Review by moonming — APPROVED** [review]

**Pull request merged**